src/golang/cmd/server/handler/get_workflow_history.go

```diff
@@ -0,0 +1,119 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aqueducthq/aqueduct/cmd/server/routes"
+	aq_context "github.com/aqueducthq/aqueduct/lib/context"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/errors"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/aqueducthq/aqueduct/lib/repos"
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+)
+
+// Route: /workflow/{workflowId}/history
+// Method: GET
+// Params:
+//  `workflowId`: the UUID of the workflow whose history we're retrieving
+// Request:
+//	Headers:
+//		`api-key`: user's API Key
+// Response:
+//	Body:
+//		serialized `workflowHistoryResponse`, a list of all the versions of this workflow, their timestamps, and their statuses
+
+type getWorkflowHistoryResponse struct {
+	Id       uuid.UUID                 `json:"id"`
+	Name     string                    `json:"name"`
+	Versions []workflowVersionResponse `json:"versions"`
```

Comment:
Since we're documenting the request, it might be good to document our responses too. This route is fairly straightforward but we should definitely cover assumptions like what is the ordering of the versions in the list.